docs/diagrams/catchup-procedure.puml

```diff
@@ -0,0 +1,317 @@
+        Node2 ->> Node4: CONSISTENCY_PROOF
+        Node3 ->> Node4: CONSISTENCY_PROOF
+        note right Node4
+            <b>CONSISTENCY_PROOF contains:
+                - <b>seq_no_start</b> <i>(seq_no of last transaction - i.e. ledger size - on lagged node),
+                - <b>seq_no_end</b> <i>(seq_no of last transaction - i.e. ledger size - on sending node),
+                - <b>(view_no, pre_prepare_seq_no)</b><i>(key of Master's 3PC batch
+                  <i>which contained last transaction on sending node),
+                - <b>old merkle root</b>  <i>(merkle root of ledger on lagged node),
+                - <b>new merkle root</b>  <i>(merkle root of ledger on sending node),
+                - <b>hashes</b> <i>(consistency proof for transactions lacking on lagged node)
+        end note
+
+        opt Did not gather f+1 same CONSISTENCY_PROOFs
+
+            Node4 -> Node4: Calculate medium other's ledger size
+            note right Node4
+                Calculate  <b>medium other's ledger size</b>
+                over received CONSISTENCY_PROOFs
+            end note
+
+            Node4 -> Node4: Discard received CONSISTENCY_PROOFs
+
+            Node4 ->> Node1: MESSAGE_REQUEST(CONSISTENCY_PROOF)
+            Node4 ->> Node2: MESSAGE_REQUEST(CONSISTENCY_PROOF)
+            Node4 ->> Node3: MESSAGE_REQUEST(CONSISTENCY_PROOF)
+            note right Node4
+                Ask other nodes for  <b>CONSISTENCY_PROOFs from
+                <b>own ledger size to medium other's ledger size
+            end note
+
+            Node1 ->> Node4: MESSAGE_RESPONSE(CONSISTENCY_PROOF)
+            Node2 ->> Node4: MESSAGE_RESPONSE(CONSISTENCY_PROOF)
+            Node3 ->> Node4: MESSAGE_RESPONSE(CONSISTENCY_PROOF)
+            note right Node4
+                Gather at least f+1  <b>same CONSISTENCY_PROOFs</b>
+            end note
+
+        end opt
+
+        note over Node1, Node4
+            Node gathered at least f+1 same CONSISTENCY_PROOFs
+        end note
+
+        Node4 -> Node4: Process stashed ORDEREDs
+        note right Node4
+            <b>Process accumulated stashed ORDEREDs:
+            For each stashed ORDERED:
+                - re-apply 3PC batch,
+                - execute 3PC batch
+        end note
+
+        alt Ledger is pool
+            Node4 -> Node4: mode = discovering
+        else Ledger is not pool
+            Node4 -> Node4: mode = syncing
+        end alt
+
+        Node4 -> Node4: Designate target ledger size for catch-up
+        note right Node4
+            Designate  <b>target ledger size</b> for catch-up
+            according to  <b>quorumed CONSISTENCY_PROOF</b>
+        end note
+
+        Node4 -> Node4: Generate CATCHUP_REQs
+        note right Node4
+            Generate <b>CATCHUP_REQs</b>  for different slices of lacking part
+            of ledger in quantity corresponding to count of other nodes
+        end note
+
+        Node4 ->> Node1: CATCHUP_REQ
```

Comment:
It would be great to give an example here, like
- we need to catch-up 40 txns with seq_no from 22 to 62
- ask Node1 for 10 txns from 22 to 32
- ask Node2 for 10 txns from 32 to 42
etc.